fontbe/src/metrics_and_limits.rs

```diff
@@ -0,0 +1,204 @@
+//! Generates the [hmtx](https://learn.microsoft.com/en-us/typography/opentype/spec/hmtx) and
+//! [hhea](https://learn.microsoft.com/en-us/typography/opentype/spec/hhea) tables
+
+use std::cmp::{max, min};
+
+use fontdrasil::orchestration::Work;
+use read_fonts::types::FWord;
+use write_fonts::{
+    dump_table,
+    tables::{hhea::Hhea, hmtx::Hmtx, vmtx::LongMetric},
+    OtRound,
+};
+
+use crate::{
+    error::Error,
+    orchestration::{BeWork, Bytes, Context, Glyph},
+};
+
+struct MetricAndLimitWork {}
+
+pub fn create_metric_and_limit_work() -> Box<BeWork> {
+    Box::new(MetricAndLimitWork {})
+}
+
+#[derive(Debug, Default)]
+struct GlyphLimits {
+    min_left_side_bearing: Option<i16>,
+    min_right_side_bearing: Option<u16>,
+    x_max_extent: Option<i16>,
+    advance_width_max: Option<u16>,
+    max_points: Option<u16>,
+    max_contours: Option<u16>,
+    max_component_elements: Option<u16>,
+}
+
+impl GlyphLimits {
+    fn update(&mut self, advance: u16, glyph: &Glyph) {
+        // min side bearings are only for non-empty glyphs
+        // we will presume only simple glyphs with no contours are empty
+        if glyph.is_empty() {
+            return;
+        }
+
+        let bbox = glyph.bbox();
+        let left_side_bearing = bbox.x_min;
+        // aw - (lsb + xMax - xMin) ... but if lsb == xMin then just advance - xMax?
+        let right_side_bearing = advance.saturating_sub(bbox.x_max.try_into().unwrap());
+        self.min_left_side_bearing = self
+            .min_left_side_bearing
+            .map(|v| min(v, left_side_bearing))
+            .or(Some(left_side_bearing));
+        self.min_right_side_bearing = self
+            .min_right_side_bearing
+            .map(|v| min(v, right_side_bearing))
+            .or(Some(right_side_bearing));
+        self.x_max_extent = self
+            .x_max_extent
+            .map(|v| max(v, bbox.x_max))
+            .or(Some(bbox.x_max));
+        self.advance_width_max = self
+            .advance_width_max
+            .map(|v| max(v, advance))
+            .or(Some(advance));
+
+        match glyph {
+            Glyph::Simple(simple) => {
+                let max_points = simple
+                    .contours()
+                    .iter()
+                    // TODO: this is stupid, just let me see the damn points
+                    .map(|c| c.clone().into_iter().count())
+                    .max()
+                    .unwrap_or_default() as u16;
+                self.max_points = self
+                    .max_points
+                    .map(|v| max(v, max_points))
+                    .or(Some(max_points));
```

Comment:
I might write this as,
```suggestion
                self.max_points = Some(self.max_points.unwrap_or_default().max(max_points));
```

Reply:
This became nicer by not using Option per your other suggestion